### PR TITLE
Refactor test detection and enhance file type handling

### PR DESCRIPTION
**Pull request by hawkeyexl**
Refactor the test detection logic and improve file type management. Add line and location tracking for detected tests, ensuring accurate detection across various content formats. Enhance the test suite with additional cases for better coverage.

## Summary by CodeRabbit

* **New Features**
  * Automatic file-type detection from content with bundled presets (Markdown, AsciiDoc, HTML, DITA).
  * Detection now attaches location metadata (line, start/end indices) and contentPath to tests/steps.
  * Added JSON/YAML spec parsing and a new public parseTests export for improved spec handling and location extraction.

* **Tests**
  * Large expansion of fixtures and coverage for detection, location tracking, ignore/inline markers, security guards, and edge cases.

* **Chores**
  * Build scripts updated to run a clean step and include test coverage in the pipeline.

### CODE REVIEW

File: src/common/src/fileTypes.ts [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+                "<\\?doc-detective\\s+test\\s+ignore\\s+end\\s*\\?>",
+                "<!--\\s*test ignore\\s+end\\s*-->",
+            ],
+            step: [
+                "<\\?doc-detective\\s+step\\s+([\\s\\S]*?)\\s*\\?>",
+                "<!--\\s*step([\\s\\S]+?)-->",
+                '<data\\s+name="step"\\s*>([\\s\\S]*?)<\\/data>',
+            ],
+        },
+        markup: [
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply src/common/src/fileTypes.ts]: Acknowledged, but not acting on this. The indentation style in this file is pre-existing and consistent within itself. A full reformatting would create a large diff unrelated to this PR's scope.

File: test/spec-location.test.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+import os from "node:os";
+import assert from "node:assert/strict";
+import { parseTests } from "../dist/core/detectTests.js";
+
+const tmpDir = os.tmpdir();
+
+function writeTempFile(name, content) {
+  const filePath = path.join(tmpDir, name);
+  fs.writeFileSync(filePath, content, "utf8");
+  return filePath;
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply test/spec-location.test.js]: Fixed. Tests now use `fs.mkdtempSync` to create a unique temp directory per run, and an `after()` hook cleans it up. No more collision risk with parallel test runs.

File: test/spec-location.test.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        const specs = await parseTests({ config: minimalConfig, files: [filePath] });
+        assert.equal(specs.length, 1);
+        const test0Step = specs[0].tests[0].steps[0];
+        const test1Step = specs[0].tests[1].steps[0];
+
+        assert.ok(test0Step.location, "Test 0 step should have location");
+        assert.ok(test1Step.location, "Test 1 step should have location");
+        assert.ok(test1Step.location.startIndex > test0Step.location.startIndex,
+          "Test 1 step should be after test 0 step in the file");
+      } finally {
```
[the remark this answers is not in the document]
Comment (hawkeyexl) [reply test/spec-location.test.js]: Added in 4b347a3 — new test "should apply location to original steps (not before-steps) and not leak _fromBefore" in spec-location.test.js. It creates a before-spec with a setup step, a main spec referencing it, then asserts: (1) no `_fromBefore` markers on any step, (2) locations are applied to the correct original steps after the prepended before-step offset.
